core/src/avm2/globals/flash/geom/perspective_projection.rs

```diff
@@ -40,7 +40,7 @@ pub fn get_focal_length<'gc>(
     let fov = this.get_slot(pp_slots::FOV).coerce_to_number(activation)?;
 
     let width = get_width(activation, this);
-    let focal_length = (width / 2.0) / f64::tan(fov / 2.0 * DEG2RAD);
+    let focal_length = (width / 2.0) as f32 * f64::tan((PI - fov * DEG2RAD) / 2.0) as f32;
```

Comment:
After many trials and observations, I found this code. It doesn't seem to be perfect yet (for edge values like 1e-10), though.

* `1 / tan x` = `cot x` = `tan (pi/2 - x)`
* `_ as f32 * f32::tan(_)` is not good. This produces better result.

Reply:
When we're much closer to FP inaccuracy thresholds, did you also try other tricks like replacing `* DEG2RAD` by explicit `* PI / 180.0`? Since doing the multiplication and division separately could also affect the result.

Reply:
I tried other cases like `_ * PI / 180.0` and `_ / 180.0 * PI`, and also both Rust `std::f64::const::PI` and ActionScript `Math.PI` defined in `Math.as`, but as far as I tested, they didn't show significant improvement...